tsdb/wal/wal_test.go

```diff
@@ -454,8 +454,9 @@ func TestLogPartialWrite(t *testing.T) {
 
 	for testName, testData := range tests {
 		t.Run(testName, func(t *testing.T) {
-			dirPath, err := ioutil.TempDir("", "")
+			dirPath, err := ioutil.TempDir("", "logpartialwrite")
 			require.NoError(t, err)
+			defer func(dir string) { require.NoError(t, os.RemoveAll(dir)) }(dirPath)
```

Comment:
```suggestion
			defer func() { require.NoError(t, os.RemoveAll(dirPath)) }()
```

Reply:
IIRC, this doesn't work, since `func() { require.NoError(t, os.RemoveAll(dirPath)) }()` dirPath will not be parsed at this moment.  In this case, we will call `defer func() { require.NoError(t, os.RemoveAll(dirPath)) }()` mutiple times, while may lead to only the last dirPath will be cleaned up.

Reply:
dirPath is a local variable in this context, so it is not carried to the next iteration. Can you try?

Reply:
And see if it works. Else this one is fine too.

Reply:
> dirPath is a local variable in this context, so it is not carried to the next iteration. Can you try?

Just checked again. your suggested change is good. thanks a lots.